src/lib/y2packager/selfupdate_addon_repo.rb

```diff
@@ -0,0 +1,65 @@
+
+
+require "fileutils"
+require "uri"
+
+require "yast"
+require "y2packager/selfupdate_addon_filter"
+require "packages/package_downloader"
+
+Yast.import "Directory"
+Yast.import "Pkg"
+
+module Y2Packager
+  # create a local add-on repository from a self-update repository
+  class SelfupdateAddonRepo
+    extend Yast::Logger
+
+    REPO_PATH = File.join(Yast::Directory.vardir, "self_update_addon").freeze
+
+    #
+    # Create an addon repository from the self-update repository
+    # containing specific packages. The repository is a plaindir type
+    # and does not contain any metadata.
+    #
+    # @param repo_id [Integer] repo_id repository ID
+    # @param path [String] path where to download the packages
+    #
+    # @return [Boolean] true if a repository has been created,
+    #   false when no addon package was found in the self update repository
+    #
+    def self.copy_packages(repo_id, path = REPO_PATH)
+      pkgs = SelfupdateAddonFilter.packages(repo_id)
+      return false if pkgs.empty?
+
+      log.info("Addon packages to download: #{pkgs}")
+
+      ::FileUtils.mkdir_p(path)
+
+      pkgs.each do |pkg|
+        downloader = Packages::PackageDownloader.new(repo_id, pkg)
+        log.info("Downloading package #{pkg}...")
+        downloader.download(File.join(path, "#{pkg}.rpm"))
+      end
+
+      log.info("Downloaded packages: #{Dir["#{path}/*"]}")
+
+      true
+    end
+
```

Comment:
It is almost self explanatory and tested, having a method documentation will make it perfect ;)